tutorials/tts/FastPitch_MultiSpeaker_Pretraining.ipynb

```diff
@@ -0,0 +1,867 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!mkdir -p {confdir} && cd {confdir} && wget https://raw.githubusercontent.com/NVIDIA/NeMo/$BRANCH/examples/tts/conf/fastpitch_align_44100_adapter.yaml"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e82e6860",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!mkdir -p {codedir} && cd {codedir} && wget https://raw.githubusercontent.com/NVIDIA/NeMo/$BRANCH/examples/tts/fastpitch.py"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "12307264",
+   "metadata": {},
+   "source": [
+    "### Important notes\n",
+    "* [Data] **speaker_id** in **sup_data_types**: each data has an unique speaker index (start from 0) in the input.\n",
+    "* [Data] **reference_audio** in **sup_data_types**: each data has a reference audio (from the same speaker) in the input.\n",
+    "* [Speaker] **model.speaker_encoder.lookup_module**: model creates lookup table to get speaker embedding from speaker id.\n",
+    "* [Speaker] **model.speaker_encoder.lookup_module.n_speakers**: model gets the speaker size. \n",
+    "* [Speaker] **model.speaker_encoder.gst_module**: model creates global style token to extract speaker information from reference audio.\n",
+    "* [Condition] **condition_types=\"['add', 'layernorm']\"**: insert conditions with `add` operation to inputs and `layernorm` operation to layernorms."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6ffdce37",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Normally 200 epochs\n",
+    "!(python {codedir}/fastpitch.py \\\n",
+    "--config-path={os.path.abspath(confdir)} \\\n",
+    "--config-name=fastpitch_align_44100_adapter.yaml \\\n",
+    "+init_from_pretrained_model=\"tts_en_fastpitch\" \\\n",
+    "sample_rate={sample_rate} \\\n",
+    "train_dataset={train_manifest} \\\n",
+    "validation_datasets={valid_manifest} \\\n",
+    "sup_data_types=\"['align_prior_matrix', 'pitch', 'speaker_id','reference_audio']\" \\\n",
+    "sup_data_path={suppdir} \\\n",
+    "pitch_mean={pitch_mean} \\\n",
+    "pitch_std={pitch_std} \\\n",
+    "phoneme_dict_path={normdir}/cmudict-0.7b_nv22.10 \\\n",
+    "heteronyms_path={normdir}/heteronyms-052722 \\\n",
+    "model.speaker_encoder.lookup_module._target_=\"nemo.collections.tts.modules.submodules.SpeakerLookupTable\" \\\n",
+    "model.speaker_encoder.lookup_module.n_speakers=5 \\\n",
+    "model.speaker_encoder.gst_module._target_=\"nemo.collections.tts.modules.submodules.GlobalStyleToken\" \\\n",
+    "model.input_fft.condition_types=\"['add', 'layernorm']\" \\\n",
+    "model.output_fft.condition_types=\"['add', 'layernorm']\" \\\n",
+    "model.duration_predictor.condition_types=\"['add', 'layernorm']\" \\\n",
+    "model.pitch_predictor.condition_types=\"['add', 'layernorm']\" \\\n",
+    "model.alignment_module.condition_types=\"['add']\" \\\n",
+    "model.train_ds.dataloader_params.batch_size=8 \\\n",
```

Comment:
Is pretraining supposed to use such a small batch size (usually we do 32)? Or is 8 only intended for fine-tuning?

Reply:
I use 8 because it is just an example. User can change based on their preference. I'll add this in the comment.